app/android/runtime_client/src/org/xwalk/app/runtime/extension/ExtensionInstanceHelper.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.app.runtime.extension;
+
+import org.json.JSONObject;
+
+public class ExtensionInstanceHelper {
+    private BindingObjectStore mStore;
+    XWalkExtensionClient mExtension;
+    MessageHandler mHandler;
+    int mId;
+
+    public ExtensionInstanceHelper(XWalkExtensionClient extension, int id) {
+        mId = id;
+        mExtension = extension;
+        // Extend the handlers from extension object to each instance.
+        // For the case, that the developer register the handers by hand
+        // to the extenson itself.
```

Comment:
Thanks for comments. I changed the words. It means copying all the handlers registered to the extension object to each instance, as we have only extension object exposed to developer currently.